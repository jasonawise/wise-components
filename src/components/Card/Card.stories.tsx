import React from "react";
import Card from "./Card";

export default {
  title: "Components/Card",
  component: Card,
};

export const Default = {
  args: {
    children: "Hello world",
    cardWidth: "w-1/2",
    customCss: ""
  },
};
