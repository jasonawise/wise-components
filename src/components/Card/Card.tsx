import { ReactNode } from "react";

interface CardProps {
  children: ReactNode;
  cardWidth?: string;
  customCss?: string;
}

//
// This card has a default width of 100%
// If you want to change the width, you can pass a custom width OR keep the width 100% and wrap a container around it
// The customCss prop is for any additional CSS you want to add to the card
//

/**
 * Card component that renders a styled div with optional custom CSS and width.
 *
 * @param {React.ReactNode} children - The content to be displayed inside the card.
 * @param {string} customCss - Additional custom CSS classes to apply to the card.
 * @param {string} [cardWidth="w-full"] - The width of the card, default is "w-full".
 * @returns {JSX.Element} The rendered card component.
 */
export default function Card({
  children,
  customCss,
  cardWidth = "w-full",
}: CardProps): JSX.Element {
  return (
    <div
      className={`border border-solid rounded-md p-4 ${cardWidth} ${customCss}`}
    >
      {children}
    </div>
  );
}
