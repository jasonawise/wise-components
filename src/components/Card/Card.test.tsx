import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";
import "@testing-library/jest-dom";

describe("Card component", () => {
  it("renders children correctly", () => {
    const { getByText } = render(<Card>Test Content</Card>);
    expect(getByText("Test Content")).toBeInTheDocument();
  });

  it("applies default width class", () => {
    const { container } = render(<Card>Test Content</Card>);
    expect(container.firstChild).toHaveClass("w-full");
  });

  it("applies custom width class", () => {
    const { container } = render(<Card cardWidth="w-1/2">Test Content</Card>);
    expect(container.firstChild).toHaveClass("w-1/2");
  });

  it("applies custom CSS classes", () => {
    const { container } = render(
      <Card customCss="bg-red-500">Test Content</Card>
    );
    expect(container.firstChild).toHaveClass("bg-red-500");
  });

  it("applies both custom width and custom CSS classes", () => {
    const { container } = render(
      <Card cardWidth="w-1/2" customCss="bg-red-500">
        Test Content
      </Card>
    );
    expect(container.firstChild).toHaveClass("w-1/2 bg-red-500");
  });
});
it("matches snapshot with default props", () => {
  const { asFragment } = render(<Card>Snapshot Test Content</Card>);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with custom width", () => {
  const { asFragment } = render(
    <Card cardWidth="w-1/2">Snapshot Test Content</Card>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with custom CSS", () => {
  const { asFragment } = render(
    <Card customCss="bg-red-500">Snapshot Test Content</Card>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with custom width and custom CSS", () => {
  const { asFragment } = render(
    <Card cardWidth="w-1/2" customCss="bg-red-500">
      Snapshot Test Content
    </Card>
  );
  expect(asFragment()).toMatchSnapshot();
});
