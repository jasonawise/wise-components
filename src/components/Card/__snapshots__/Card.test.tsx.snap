// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="border border-solid rounded-md p-4 w-full undefined"
  >
    Snapshot Test Content
  </div>
</DocumentFragment>
`;
